src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -827,9 +836,13 @@ private void OutputToOutputWindow(string message)
                     continue;
                 }
 
-                projectContext.AddMetadataReference(path, reference.Properties);
+                if (addedReferencePaths.Add(path))
```

Comment:
We do a fake resolve on our side (resolve to the project file) - which is broken (https://github.com/dotnet/project-system/issues/1872),